File: leetcode/0875-KokoEatingBananas/solution.py
from typing import List
import math

class Solution:
    def minEatingSpeed(self, piles: List[int], h: int) -> int:
        low, high = 1, max(piles) # range of possible k values
        result = 0
        # use binary search to try different k values
        while low<=high:
            mid = low + (high-low)//2
            # find total hours it would take to eat with k
            total_time = 0
            for pile in piles:
                total_time += math.ceil(float(pile)/mid)
            # binary search logic
            if total_time<=h:
                result = mid
                high = mid-1
            else:
                low = mid+1
        return result
    
print(Solution().minEatingSpeed([3,6,7,11],8)) # expected output: 4

File: leetcode/0875-KokoEatingBananas/solution.ts
function minEatingSpeed(piles: number[], h: number): number {
    let min_k = 1, max_k = Math.max(...piles);
    let result = 0;
    
    while (min_k <= max_k){
        let candidate_k = min_k + Math.floor((max_k-min_k)/2)
        let time_taken = 0;
        for (let pile of piles){
            time_taken += Math.ceil(pile/candidate_k);
        }
        if (time_taken>h){
            min_k = candidate_k + 1
        }
        else{
            max_k = candidate_k - 1
            result = candidate_k
        }

    }


    return result
};t